vtm/src/org/oscim/renderer/bucket/ExtrusionBuckets.java

```diff
@@ -33,22 +52,98 @@ public ExtrusionBuckets(MapTile tile) {
     }
 
     /**
-     * Set new ExtrusionLayers and clear previous.
+     * Add ExtrusionBucket at last position
      */
-    public void setBuckets(ExtrusionBucket el) {
+    private void add(ExtrusionBucket el) {
+        if (buckets == null) {
+            this.buckets = el;
+        }
+        RenderBucket b = buckets;
+        while (b.next != null) b = b.next;
+
+        b.next = el;
```

Comment:
(in any case)
If `this.buckets = el` then where `el` is attached, to itself?

Reply:
Always these mistakes :see_no_evil: Missed a return of course :grin:
But can omit it anyway...